tor-android-binary/src/main/java/org/torproject/jni/TorService.java

```diff
@@ -377,6 +394,7 @@ public void run() {
                         mainConfigurationFree();
                         Log.i(TAG, "Releasing lock");
                         runLock.unlock();
+                        if (owningProcessPid != -1) stopSelf();
```

Comment:
One could also consider to always stop here. What use is the service if Tor stopped?

I just made this conditional to not change existing behavior.

Reply:
Yeah, it does seem like the `finally` block should call `stopSelf`.  I think the `try`/`finally` block should be unnested and combined with the `catch (IllegalStateException | IllegalArgumentException | InterruptedException` `try` block.  @akwizgran any reason in particular why you made it nested in #59?

Reply:
Here's an attempt at implementing this https://github.com/guardianproject/tor-android/pull/65

Reply:
@eighthave as far as I can see it's fine to un-nest the try block.